### PR TITLE
Feat rspy282/document stac catalog metadata

### PR DESCRIPTION
**Pull request by agrosu1978**
Update the stac catalog metadata by using environment variables. In cluster running case, the values for these env vars should be set in the values.yaml file

### CODE REVIEW

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 1]
```diff
@@ -567,6 +567,21 @@ def update_links_for_all_collections(self, collections: list[dict]) -> list[dict
                 link["href"] = link_parser._replace(path=new_path).geturl()
         return collections
 
+    def update_stac_catalog_metadata(self, metadata: dict):
+        """Update the metadata fields from a catalog, following the US RSPY-282
```
Comment (Padeanu) [inline services/catalog/rs_server_catalog/user_catalog.py]: Remove private JIRA references from comments.

File: services/catalog/rs_server_catalog/user_catalog.py [hunk 2]
```diff
@@ -567,6 +567,21 @@ def update_links_for_all_collections(self, collections: list[dict]) -> list[dict
...
+
+        Args:
+            metadata (dict): The metdata that has to be updated. The fields id, title,
+                            description and stac_version are to be updated, by using the env vars which have
+                            to be set before starting the app/pod. The existing values are used if
+                            the env vars are not found
+        """
+        if metadata.get("type") != "Catalog":
```
Comment (Padeanu) [inline services/catalog/rs_server_catalog/user_catalog.py]: Why not just

```python
if metadata.get("type") == "Catalog":
   for ..
```

File: services/catalog/tests/test_endpoints.py [hunk 3]
```diff
@@ -79,6 +79,40 @@ def test_status_code_200_docs_if_good_endpoints(client):  # pylint: disable=miss
     assert response.status_code == fastapi.status.HTTP_200_OK
 
 
+def test_update_stac_catalog_metadata(client):
+    """
+    Test the update od the stac catalog metdata when the /catalog/ endpoint is called
```
Comment (Padeanu) [inline services/catalog/tests/test_endpoints.py]: typo

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 1 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py]: fixed

File: services/catalog/rs_server_catalog/user_catalog.py (hunk 2 above)
Comment (agrosu1978) [reply services/catalog/rs_server_catalog/user_catalog.py]: fixed (personal point of view)

**Review by Padeanu — APPROVED** [review]
lgtm

**Review by cyouri-csgroup — APPROVED** [review]

**Review by jgaucher-cs — APPROVED** [review]

**Pull request merged**